### ISSUE TITLE
mono-cil-strip doesn't sort the custom metadata table

### ISSUE THREAD

**Issue by rolfbjarne** [mono/mono#21302]
Test assembly: [Xamarin.TVOS.dll.zip](https://github.com/mono/mono/files/7516082/Xamarin.TVOS.dll.zip)

Repro:
* Download & extract test assembly
* Execute `mono-cil-strip Xamarin.TVOS.dll Xamarin.TVOS-stripped.dll`
* Notice that the metadata tables in `Xamarin.TVOS-stripped.dll' are not sorted. This program can verify this: https://github.com/dotnet/runtime/issues/61402#issuecomment-965630534

**Comment by marek-safar** [comment]
Is this for pre-net6 use cases only?

**Comment by rolfbjarne** [comment]
We use mono-cil-strip for both pre-net6 and .NET 6, so it's for both.

**Comment by akoeplinger** [comment]
@rolfbjarne mono-cil-strip should only be used in the "strip IL for FullAOT" context, do you see any issue there too? just trying to understand how important this is

**Comment by marek-safar** [comment]
I think we need to fix this in dotnet/runtime because we use the same logic (and Cecil) there so this is impacting .NET6

**Comment by rolfbjarne** [comment]
> @rolfbjarne mono-cil-strip should only be used in the "strip IL for FullAOT" context, do you see any issue there too? just trying to understand how important this is

Yes, it's an issue for FullAOT as well, because attributes are stored in the managed assembly, so any attribute lookup might fail to find the attribute even though it's there.
